samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -366,8 +378,8 @@ private TaskFactory createTaskFactory() {
     }
     InMemorySystemDescriptor imsd = (InMemorySystemDescriptor) descriptor.getSystemDescriptor();
     imsd.withInMemoryScope(this.inMemoryScope);
-    addConfigs(descriptor.toConfig());
-    addConfigs(descriptor.getSystemDescriptor().toConfig());
+    putIfAbsentConfig(descriptor.toConfig());
```

Comment:
`putIfAbsentConfig` API here is entirely unnecessary. You had to do this, since we didn't use the StreamMessageType as a type for the API signature(rather used a raw type). Rather than the following(at line 369): 

```java
 private <StreamMessageType> void initializeInMemoryInputStream(InMemoryInputDescriptor descriptor,
       Map<Integer, Iterable<StreamMessageType>> partitonData) {	       Map<Integer, Iterable<StreamMessageType>> partitonData) {
```

We can do the following :

```java
 private <StreamMessageType> void initializeInMemoryInputStream(InMemoryInputDescriptor<? extends StreeamMessageType> descriptor,
       Map<Integer, Iterable<StreamMessageType>> partitonData) {	       Map<Integer, Iterable<StreamMessageType>> partitonData) {
```

Since we haven't explicitly defined the type for genericly parameterized class, type-erasure kicks in and all the methods from `InMemoryInputDescriptor` return response of object type. If you do the above, then you don't have to do explicit typecast in all the places(For instance, at line 372 from `Object` to `String`).

After the above API signature change  `putifAbsentConfig` API will be unnecessary, and could be simply in-lined it with the following: 

```java
    descriptor.toConfig().forEach(this.configs::putIfAbsent);
```

If it does makes sense, then please change it.